spec/app/db/repository/authenticator_repository_spec.rb

```diff
@@ -42,6 +42,210 @@
     end
   end
 
+  describe("find_all") do
+    context "missing parameters" do
+      it "returns nil with no parameters" do
+        expect(repo.find_all(type: nil, account: nil)).to eq([])
+      end
+
+      it "returns nil with no type parameter" do
+        expect(repo.find_all(type: nil, account: "rspec")).to eq([])
+      end
+
+      it "returns nil with no account parameter" do
+        expect(repo.find_all(type: "oidc", account: nil)).to eq([])
+      end
+
+      it "returns nil with no service_id parameter" do
+        expect(repo.find_all(type: "oidc", account: "rspec")).to eq([])
+      end
+    end
+
+    context "no variables are set" do
+      it "returns nil when the authenticator doesn't exist" do
+        expect(repo.find_all(type: "oidc", account: "rspec")).to eq([])
+      end
+    end
+
+    context "no variables are set" do
+      before do
+        ::Role.create(
+          role_id: "rspec:policy:conjur/authn-oidc/abc123"
+        )
+        ::Resource.create(
+          resource_id: "rspec:webservice:conjur/authn-oidc/abc123",
+          owner_id: "rspec:policy:conjur/authn-oidc/abc123"
+        )
+      end
+
+      it "returns a v2 authenticator with all properties set" do
```

Comment:
[Note] Looks like these tests might be related to Eric's original work.  I don't see the V2 data object in this PR.

Reply:
yeah. i think the authenticator class he created handles that